incubator/binding-openapi/src/main/java/io/aklivity/zilla/runtime/binding/openapi/internal/config/OpenapiOptionsConfigAdapter.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc
+ *
+ * Licensed under the Aklivity Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.binding.openapi.internal.config;
+
+import static org.agrona.LangUtil.rethrowUnchecked;
+
+import java.util.function.Function;
+
+import jakarta.json.Json;
+import jakarta.json.JsonArrayBuilder;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonObjectBuilder;
+import jakarta.json.JsonString;
+import jakarta.json.JsonValue;
+import jakarta.json.bind.Jsonb;
+import jakarta.json.bind.JsonbBuilder;
+import jakarta.json.bind.adapter.JsonbAdapter;
+
+import io.aklivity.zilla.runtime.binding.http.config.HttpOptionsConfig;
+import io.aklivity.zilla.runtime.binding.openapi.config.OpenapiConfig;
+import io.aklivity.zilla.runtime.binding.openapi.config.OpenapiOptionsConfig;
+import io.aklivity.zilla.runtime.binding.openapi.config.OpenpaiOptionsConfigBuilder;
+import io.aklivity.zilla.runtime.binding.openapi.internal.OpenapiBinding;
+import io.aklivity.zilla.runtime.binding.openapi.internal.model.OpenApi;
+import io.aklivity.zilla.runtime.binding.tls.config.TlsOptionsConfig;
+import io.aklivity.zilla.runtime.engine.config.ConfigAdapterContext;
+import io.aklivity.zilla.runtime.engine.config.OptionsConfig;
+import io.aklivity.zilla.runtime.engine.config.OptionsConfigAdapter;
+import io.aklivity.zilla.runtime.engine.config.OptionsConfigAdapterSpi;
+
+public final class OpenapiOptionsConfigAdapter implements OptionsConfigAdapterSpi, JsonbAdapter<OptionsConfig, JsonObject>
+{
+    private static final String TLS_NAME = "tls";
+    private static final String HTTP_NAME = "http";
+    private static final String SPECS_NAME = "specs";
+    private OptionsConfigAdapter optionsAdapter;
```

Comment:
```suggestion
    private OptionsConfigAdapter tlsOptions;
    private OptionsConfigAdapter httpOptions;
```

and a constructor
```
  public OpenapiOptionsConfigAdapter()
  {
      tlsOptions.adaptType("tls");
      httpOptions.adaptType("http");
  }
```

Reply:
I guess you mean in `public void adaptContext` since there you have access to context